src/Security/Authentication/OpenIdConnect/src/OpenIdConnectConfigureOptions.cs

```diff
@@ -0,0 +1,136 @@
+    }
+
+    private static CookieBuilder GetCookieFromConfig(IConfiguration cookieConfigSection, CookieBuilder cookieBuilder)
+    {
+        if (cookieConfigSection is not null && cookieConfigSection.GetChildren().Any())
+        {
+            // Override the existing defaults when values are set instead of constructing
+            // an entirely new CookieBuilder.
+            cookieBuilder.Domain = cookieConfigSection[nameof(cookieBuilder.Domain)];
+            cookieBuilder.HttpOnly = StringHelpers.ParseValueOrDefault(cookieBuilder.HttpOnly, cookieConfigSection[nameof(cookieBuilder.HttpOnly)], bool.Parse);
+            cookieBuilder.IsEssential = StringHelpers.ParseValueOrDefault(cookieBuilder.IsEssential, cookieConfigSection[nameof(cookieBuilder.IsEssential)], bool.Parse);
+            cookieBuilder.Expiration = cookieBuilder.Expiration is TimeSpan expiration ? StringHelpers.ParseValueOrDefault(expiration, cookieConfigSection[nameof(cookieBuilder.Expiration)], _invariantTimeSpanParse) : cookieBuilder.Expiration;
+            cookieBuilder.MaxAge = cookieBuilder.MaxAge is TimeSpan maxAge ? StringHelpers.ParseValueOrDefault(maxAge, cookieConfigSection[nameof(cookieBuilder.MaxAge)], _invariantTimeSpanParse) : cookieBuilder.MaxAge;
+            cookieBuilder.Name = cookieConfigSection[nameof(CookieBuilder.Name)] ?? cookieBuilder.Name;
+            cookieBuilder.Path = cookieConfigSection[nameof(CookieBuilder.Path)];
+            cookieBuilder.SameSite = cookieConfigSection[nameof(CookieBuilder.SameSite)]?.ToLowerInvariant() switch
+            {
+                "lax" => SameSiteMode.Lax,
+                "strict" => SameSiteMode.Strict,
+                "unspecified" => SameSiteMode.Unspecified,
+                "none" => SameSiteMode.None,
+                null => cookieBuilder.SameSite,
+                "" => cookieBuilder.SameSite,
+                _ => throw new InvalidOperationException($"{nameof(CookieBuilder.SameSite)} option must be a valid {nameof(SameSiteMode)}")
+            };
+            cookieBuilder.SecurePolicy = cookieConfigSection[nameof(CookieBuilder.SecurePolicy)]?.ToLowerInvariant() switch
+            {
+                "always" => CookieSecurePolicy.Always,
+                "none" => CookieSecurePolicy.None,
+                "sameasrequest" => CookieSecurePolicy.SameAsRequest,
+                null => cookieBuilder.SecurePolicy,
+                "" => cookieBuilder.SecurePolicy,
+                _ => throw new InvalidOperationException($"{nameof(CookieBuilder.SameSite)} option must be a valid {nameof(SameSiteMode)}")
+            };
+
+            var extensions = cookieConfigSection.GetSection(nameof(CookieBuilder.Extensions)).GetChildren().Select(ext => ext.Value).OfType<string>();
+            foreach (var extension in extensions)
+            {
+                cookieBuilder.Extensions.Add(extension);
```

Comment:
Should extensions be cleared before adding?

Reply:
I had a bit of a change of heart here. The Azure AD extension methods which end up overwriting list options because they use `config.Bind`. It makes sense to me that list options should align more closely with that behavior.